File: Programmin_in_Java-Erasmus/Final_Project/src/main/java/server/validators/UserValidator.java
package main.java.server.validators;

import main.java.server.domain.User;
import main.java.server.exceptions.ValidationException;

public class UserValidator extends BaseValidator<User> {
    /**
     * Checks if the user has valid email and name
     *
     * @param user the given user
     * @throws ValidationException if the user is invalid
     */
    @Override
    public void validate(User user)  {
        String text = "";
        text += validateEmail(user.getEmail()) ? "" : "Email is invalid";
        text += validateText(user.getName()) ? "" : "Name is empty";
        if (!text.isEmpty()) throw new ValidationException(text);
    }
}

File: Programmin_in_C#-Erasmus/Final_Project/Validators/UserValidator.cs
using Final_Project.Domain;
using Final_Project.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace Final_Project.Validators
{
    public class UserValidator : BaseValidator<User>
    {
        public override void validate(User user)
        {
            string text = "";
            text += validateEmail(user.getEmail()) ? "" : "Email is invalid";
            text += validateText(user.getName()) ? "" : "Name is empty";
            if (text.Length > 0)
            {
                throw new ValidationException(text);
            }
        }
    }
    
}
